java/statestore/src/main/java/sleeper/statestore/dynamodb/DynamoDBFileReferenceStore.java

```diff
@@ -127,6 +128,86 @@ public void addFiles(List<FileReference> fileReferences) throws StateStoreExcept
         }
     }
 
+    @Override
+    public void splitFileReferences(List<SplitFileReferenceRequest> splitRequests) throws StateStoreException {
+        long updateTime = clock.millis();
+        List<SplitFileReferenceRequest> batchRequests = new ArrayList<>();
+        List<TransactWriteItem> batchReferenceWrites = new ArrayList<>();
+        Map<String, Integer> batchReferenceCountUpdates = new HashMap<>();
+        int firstUnappliedRequestIndex = 0;
+        try {
+            for (int i = 0; i < splitRequests.size(); i++) {
+                SplitFileReferenceRequest splitRequest = splitRequests.get(i);
+                List<TransactWriteItem> requestWrites = splitFileReferenceWrites(splitRequest, updateTime);
+                int newBatchWrites = batchReferenceWrites.size() + requestWrites.size() + batchReferenceCountUpdates.size();
+                if (!batchReferenceCountUpdates.containsKey(splitRequest.getFilename())) {
+                    // Reference count updates need to be aggregated into one for each file, so this will only result in
+                    // a separate write item if this file has not been updated by a previous request
+                    newBatchWrites += 1;
+                }
+                if (newBatchWrites > 100) {
```

Comment:
Add comment that the reason this is 100 is that Dynamo transactions can have at most 100 items.

Reply:
Done